rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -205,171 +213,158 @@ def save_curve_data_to_cache(
-    )
-    pool_count = factory_contract.call(
-        node_inquirer=ethereum,
-        method_name='pool_count',
-    )
-    pool_addrs_result = ethereum.multicall_specific(
-        contract=factory_contract,
-        method_name='pool_list',
-        arguments=[(x,) for x in range(pool_count)],
-        decode_result=False,  # don't decode since decoding unchecksums address
-    )
-    pool_addresses = []
-    for encoded_pool_addr in pool_addrs_result:
-        pool_addresses.append(hex_or_bytes_to_address(encoded_pool_addr))
-
-    # either a pool of stablecoins or a pair `stablecoin - other curve pool token`
-    # so no need to query underlying tokens
-    pool_coins_result = ethereum.multicall_specific(
-        contract=factory_contract,
-        method_name='get_coins',
-        arguments=[(x,) for x in pool_addresses],
-    )
-    pools_mapping: CURVE_POOLS_MAPPING_TYPE = {}
-    for pool_addr, (decoded_coins_result,) in zip(pool_addresses, pool_coins_result):
-        if pool_addr in IGNORED_CURVE_POOLS:
-            continue
-        pool_coins = []
-        for coin_addr in decoded_coins_result:
-            checksumed_coin_addr = to_checksum_address(coin_addr)
-            if checksumed_coin_addr == ZERO_ADDRESS:
-                # curve returns array of fixed length. So if current pool utilizes fewer
-                # coins than the array's length, first values in the result are the real
-                # coins and all addresses after are just 0x00..00
-                break
-            pool_coins.append(checksumed_coin_addr)
-
-        # for metapools pool addr is the lp token as well
-        pools_mapping[pool_addr] = (pool_addr, pool_coins, None)
-
-    return pools_mapping
-
-
-def query_curve_gauges(
-        ethereum: 'EthereumInquirer',
-        registry_contract: 'EvmContract',
-        known_pools: list[ChecksumEvmAddress],
-) -> dict[ChecksumEvmAddress, ChecksumEvmAddress]:
-    """Queries curve gauges and returns a mapping pool address -> gauge address"""
-    pools_to_gauges: dict[ChecksumEvmAddress, ChecksumEvmAddress] = {}
-    for pool_address in known_pools:
-        gauge_address = registry_contract.call(
-            node_inquirer=ethereum,
-            method_name='get_gauges',
-            arguments=[pool_address],
-        )[0][0]
-        if gauge_address != ZERO_ADDRESS:
-            pools_to_gauges[pool_address] = to_checksum_address(gauge_address)
-
-    return pools_to_gauges
+    try:
+        pools_data = query_curve_data_from_api()
+    except CurveApiException as e:
+        log.error(f'Could not query curve api due to: {str(e)}. Will query metaregistry on chain')
+        try:
+            pools_data = query_curve_data_from_chain(ethereum)
+        except RemoteError as err:
+            log.error(f'Could not query chain for curve pools due to: {str(err)}')
+            pools_data = []  # Setting pools_data to an empty list so that it is re-queried later
```

Comment:
This could have been initialized at the top. Say before the try/except
minor detail though